### PR TITLE
bump: bump Android SDK 7.17.0

### LINKED ISSUE #357: NullPointerException in LifecycleRegistry.removeObserver
### Environment

Cordova CLI: 11.1.0
cordova-android: 13.0.0
sentry-cordova: 1.2.0, and also 1.4.0

### Steps to Reproduce

We stared receiving crash reports in sentry and within google play about a crash in our app when using sentry-cordova 1.2.0.  We updated to 1.4.0 to fix, but we appear to have the same issue.  I've attached some screenshots to describe the problem further.  In the screenshots, v1.30.7 uses sentry-cordova 1.2.0, and v1.30.8 uses sentry-cordova 1.4.0.  I cannot reproduce anything locally.

This issue looks similar to this: https://github.com/getsentry/sentry-java/issues/2675

It appears from the sentry-cordova release notes for sentry-cordova 1.4.0 that it is using v7.14.0 of the android sdk, and this supposedly has the fix.  However I still see the same message in google play, so something isn't quite right here.

### Expected Result

I would expect that updating to cordova-android would fix the problem as described by the google play error report.

### Actual Result

Image
Image
Image

### PR DESCRIPTION
**Pull request by lucas-zimerman**
Fixes #357

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+## Unreleased
+
+### Dependencies
+
+- Bump Android SDK from v7.14.0 to v7.16.0 ([#359](https://github.com/getsentry/sentry-cordova/pull/359))
+  - [changelog](https://github.com/getsentry/sentry-java/blob/7.16.0/CHANGELOG.md)
+  - [diff](https://github.com/getsentry/sentry-java/compare/7.14.0...7.16.0)
```
Comment (lucas-zimerman) [inline CHANGELOG.md:9]:
```suggestion
- Bump Android SDK from v7.14.0 to v7.17.0 ([#359](https://github.com/getsentry/sentry-cordova/pull/359))
  - [changelog](https://github.com/getsentry/sentry-java/blob/7.17.0/CHANGELOG.md)
  - [diff](https://github.com/getsentry/sentry-java/compare/7.14.0...7.17.0)
```

File: plugin.xml [hunk 2]
```diff
@@ -9,7 +9,7 @@
     <repo>https://github.com/getsentry/sentry-cordova.git</repo>
     <issue>https://github.com/getsentry/sentry-cordova/issues</issue>
 
-    <preference name="SENTRY_ANDROID_SDK_VERSION" default="7.14.0"/>
+    <preference name="SENTRY_ANDROID_SDK_VERSION" default="7.16.0"/>
```
Comment (lucas-zimerman) [inline plugin.xml:12]:
```suggestion
    <preference name="SENTRY_ANDROID_SDK_VERSION" default="7.17.0"/>
```

File: sample/package.json [hunk 3]
```diff
@@ -41,12 +41,12 @@
     },
     "plugins": {
       "sentry-cordova": {
-        "SENTRY_ANDROID_SDK_VERSION": "7.14.0"
+        "SENTRY_ANDROID_SDK_VERSION": "7.16.0"
```
Comment (lucas-zimerman) [inline sample/package.json:44]:
```suggestion
        "SENTRY_ANDROID_SDK_VERSION": "7.17.0"
```

**Review by krystofwoldrich — APPROVED** [review]
LG 🚀

**Comment by krystofwoldrich** [comment]
The yarn.lock change looks like it was not intentional.

**Comment by lucas-zimerman** [comment]
> The yarn.lock change looks like it was not intentional.

No Idea why it decided to change it.
I recreated the yarn lock and it is now using the original url now.

**Pull request merged**